test/e2e/cloud/nodes.go

```diff
@@ -0,0 +1,83 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cloud
+
+import (
+	"time"
+
+	"k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	clientset "k8s.io/client-go/kubernetes"
+	kubeadmconstants "k8s.io/kubernetes/cmd/kubeadm/app/constants"
+	"k8s.io/kubernetes/test/e2e/framework"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = SIGDescribe("Nodes [Feature:CloudProvider]", func() {
+	f := framework.NewDefaultFramework("cloudprovider")
+	var c clientset.Interface
+	BeforeEach(func() {
+		// TODO: implement for other providers
+		framework.SkipUnlessProviderIs("aws")
+		c = f.ClientSet
+	})
+
+	It("should be deleted on API server if it doesn't exist in the cloud provider", func() {
+		By("deleting a node on the cloud provider")
+		nodeOpts := metav1.ListOptions{}
+		nodes, err := c.CoreV1().Nodes().List(nodeOpts)
+		if err != nil {
+			framework.Failf("failed to list nodes: %s", err)
+		}
+
+		nodeToDelete := pickNodeNotControlPlane(nodes)
+		if nodeToDelete == nil {
+			framework.Failf("could not find node to delete, either all belong to the control plan or non exist")
+		}
+		err = framework.DeleteNodeOnCloudProvider(nodeToDelete)
+		if err != nil {
+			framework.Failf("failed to delete node %q, err: %q", nodeToDelete.Name, err)
+		}
+
+		deleted, err := framework.WaitForNodeToBeDeleted(c, nodeToDelete, time.Minute)
+		framework.ExpectNoError(err)
+		Expect(deleted).To(BeTrue())
+
+		newNodes, err := c.CoreV1().Nodes().List(nodeOpts)
+		if err != nil {
+			framework.Failf("failed to list nodes: %s", err)
+		}
+
+		Expect(len(newNodes.Items)).To(Equal(len(nodes.Items) - 1))
+	})
+})
+
+// pickNodeNotControlPlane picks the first node that is not
+// part of the control plane.
+func pickNodeNotControlPlane(nodes *v1.NodeList) *v1.Node {
```

Comment:
Does it make sense to use taints here to avoid picking nodes that might have other pods scheduled on them? You could either add a tainted node to the cluster to avoid ever having anything scheduled on it or taint an existing node with `NoExecute` and wait for everything to be evicted. WDYT?

Reply:
Marking the test disruptive might be a less complicated option.

Reply:
Agreed! Let's stick with marking it disruptive and add the taint/evict logic in the future if needed?

Reply:
SGTM